.github/workflows/docker-build.yml

```diff
@@ -29,23 +29,25 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build tag image
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build and push latest
         if: startsWith(github.ref, 'refs/tags/')
-        run: |
-          sudo apt update --yes
-          sudo apt install --yes --quiet binfmt-support qemu-user-static qemu-system
-          docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
-          docker buildx create --name mybuilder
-          docker buildx use mybuilder
-          docker buildx inspect --bootstrap
-          docker buildx build --push --platform linux/amd64,linux/arm64 --tag ghcr.io/valhalla/valhalla:${{ steps.extract_tag.outputs.tag }} .
-      - name: Build latest image
+        uses: docker/build-push-action@v6
```

Comment:
but like what is in this image? what operating system is it even?

Reply:
I don't fully understand what you're asking. This just runs the docker build inside the current runner, so nothing should change compared to what the process was before, right? It's just a shorthand for running the build steps like we did previously (including the multi-platform configuration) plus now we also get caching, which this action takes care of for us.

Reply:
ok so basically we dont have custom commands these "actions" encapsolate the commands plus hopefully add caching for us? i havent look at github actions much, just a couple of times when trying them out